### PR TITLE
Tracking: remove dependency to the Jetpack plugin

### LINKED ISSUE #18111: Use of undefined constant JETPACK__PLUGIN_FILE
#### Steps to reproduce the issue

1. Turn off Jetpack plugin
2. Include [`jetpack-tracking`](https://github.com/Automattic/jetpack-tracking) package in your project
3. Instantiate the class and enqueue JavaScript:
```php
$tracking = new Tracking();
add_action( 'admin_enqueue_scripts', array( $tracking, 'enqueue_tracks_scripts' ) );
```

#### What I expected
To be able to use Jetpack Tracking package outside the Jetpack Plugin.

#### What happened instead
PHP Warnings 😝 

Also to note - the warnings aren't the only issue here. Looks like `tracks-ajax.js` is also not a part of the Jetpack Tracking package: https://github.com/Automattic/jetpack-tracking/blob/c5cf2930f76e616c37dc366f0d1dcd42de9efcf1/src/class-tracking.php#L47-L55

So without the Jetpack plugin active, we can track with JavaScript.

Thanks 🙂

### PR DESCRIPTION
**Pull request by jeherve**
Fixes #18111

#### Changes proposed in this Pull Request:

This PR introduces a few changes:
- require Assets package in the Tracking package; we will use it to enqueue scripts from the package
- move Tracks lib inside the Tracking package
- use the new file in the Tracking package, and remove usage of the `JETPACK__VERSION` and `JETPACK__PLUGIN_FILE` constants.

#### Jetpack product discussion

* N/A
* 
#### Does this pull request change what data or activity we track or use?

* No

#### With Jetpack

1. Checkout this branch.
2. Connect Jetpack.
3. Manually delete the blog token using wp cli.
4. Restore the connection using the “Restore” button in the admin notice in wp-admin.
5. The restore process calls `Automattic\Jetpack\Connection\reconnect()`, which triggers the `jetpack_restore_connection_reconnect` tracking event.
6. Check tracks make sure the event was recorded.
7. Check the error logs for errors.

#### With Client-Example (no Jetpack)
1. Install client-example.
2. Manually move the this branch’s `jetpack-tracking` package into client-example’s `vendor/automattic` folder, replacing the `jetpack-tracking` package that’s already there.
3. Manually add a tracking event to the `Automattic\Jetpack\Connection\register()` method.
4. In wp-admin, register the site.
5. Check tracks to make sure the event was recorded.
6. Check the error logs for errors.

#### Proposed changelog entry for your changes:

* N/A

### CODE REVIEW

**Review by kbrown9 — COMMENTED** [review]
This works well for me!

### How I tested

#### With Jetpack

1. Checkout this branch.
2. Connect Jetpack.
3. Manually delete the blog token using wp cli.
4. Restore the connection using the “Restore” button in the admin notice in wp-admin.
5. The restore process calls `Automattic\Jetpack\Connection\reconnect()`, which triggers the `jetpack_restore_connection_reconnect` tracking event.
6. Check tracks make sure the event was recorded.
7. Check the error logs for errors.

#### With Client-Example (no Jetpack)
1. Install client-example.
2. Manually move the this branch’s `jetpack-tracking` package into client-example’s `vendor/automattic` folder, replacing the `jetpack-tracking` package that’s already there.
3. Manually add a tracking event to the `Automattic\Jetpack\Connection\register()` method.
4. In wp-admin, register the site.
5. Check tracks to make sure the event was recorded.
6. Check the error logs for errors.

**Comment by jeherve** [comment]
@kbrown9 Thank you! I'll steal your testing instructions and add them to the PR description :) 

@pyronaur Do you think you could give it a try with Jetpack Boost, see how that works for you?

Thank you!

**Review by kbrown9 — DISMISSED** [review]
~This should be ready to merge after Boost confirms that it's working for them.~ I just noticed the failing required check: https://github.com/Automattic/jetpack/pull/18112/checks?check_run_id=1565346795

**Comment by jeherve** [comment]
> I just noticed the failing required check: [#18112 (checks)](https://github.com/Automattic/jetpack/pull/18112/checks?check_run_id=1565346795)

Ah yes, good point. I just pushed 64a5bc3275755fcb0acdd5b66792621e780ce166 to address that.

**Review by kbrown9 — DISMISSED** [review]

**Comment by kbrown9** [comment]
The `tracks-ajax.js` file  is in the array in the `photon-cdn/jetpack-manifest.php` [file](https://github.com/Automattic/jetpack/blob/master/modules/photon-cdn/jetpack-manifest.php#L61). I suppose we also need to update that file. (Sorry I didn't notice this during my review!)

**Comment by pyronaur** [comment]
Files load successfully, no errors.

However, running the tracks function from the console triggers a `400` error when Jetpack plugin isn't active.

```
jpTracksAJAX.record_ajax_event('custom_event_name')
```

I suspect that it's because `admin-ajax` action isn't registered from the package.

**Comment by jeherve** [comment]
> The tracks-ajax.js file is in the array in the photon-cdn/jetpack-manifest.php file. 

That should be okay. This is updated as part of the release script.

**Comment by jeherve** [comment]
@pyronaur It does seem better now. Could you take another look?

image

**Review by pyronaur — APPROVED** [review]
LGTM!

File: packages/tracking/src/class-tracking.php [hunk 1]
```diff
@@ -11,6 +11,15 @@
  * The Tracking class, used to record events in wpcom
  */
 class Tracking {
+	/**
+	 * The assets version.
+	 *
+	 * @since 9.3.0
```
Comment (kbrown9) [inline packages/tracking/src/class-tracking.php:17]:
```suggestion
	 * @since 9.4.0
```

**Review by kbrown9 — DISMISSED** [review]
Everything tests well for me.

I pushed commit 36317ce to update the Jetpack version in the docblock.

**Comment by kbrown9** [comment]
I rebased on master and force pushed the commits in an attempt to resolve the failing `Composer / composer.lock is up to date` test, but that didn't resolve the failure.

**Comment by kraftbj** [comment]
I'm checking into this.

**Comment by kraftbj** [comment]
Let's see if that did it.

I think since the downstream `packages/tracking/composer.json` changed, technically, the root `composer.lock` needs to be updated to ack that new requirement.

**Comment by anomiex** [comment]
> I think since the downstream `packages/tracking/composer.json` changed, technically, the root `composer.lock` needs to be updated to ack that new requirement.

Yeah, it needs to pick up the new dependency into composer.lock.

**Comment by kraftbj** [comment]
I rebased against `master` post repo reorg. @kbrown9 Since I had to do a bit of work with the merge, would you care to validate/test this and approve it?

Thank you!

**Comment by kbrown9** [comment]
I pushed another update to `composer.lock` because of a failing test.

**Review by kbrown9 — APPROVED** [review]
This tests well for me.

**Comment by kbrown9** [comment]
I had to update `composer.json` again, and now the `JS tests / Build dashboard & extensions (pull_request)` test is failing. That failure appears to be unrelated to this PR, and I think it should be fixed by #18459.

**Review by kraftbj — APPROVED** [review]

**Comment by jeherve** [comment]
r220383-wpcom

**Pull request merged**